Assets/MixedRealityToolkit/MSBuild/Scripts/MSBuildTools.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if UNITY_EDITOR
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEditor;
+using UnityEditor.Compilation;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.MSBuild
+{
+    /// <summary>
+    /// Class that exposes the MSBuild project generation operation.
+    /// </summary>
+    public static class MSBuildTools
+    {
+        private static readonly HashSet<BuildTarget> supportedBuildTargets = new HashSet<BuildTarget>()
+        {
+            BuildTarget.StandaloneWindows,
+            BuildTarget.StandaloneWindows64,
+            BuildTarget.iOS,
+            BuildTarget.Android,
+            BuildTarget.WSAPlayer
+        };
+
+        public const string CSharpVersion = "7.3";
+        //public const string UWPTargetPlatformVersion = "10.0.18362.0";
+
+        private static readonly string uwpMinPlatformVersion = EditorUserBuildSettings.wsaMinUWPSDK;// "10.0.14393.0";
+        private static readonly string uwpTargetPlatformVersion = EditorUserBuildSettings.wsaUWPSDK;
+
+        [MenuItem("MSBuild/Generate C# SDK Projects")]
+        public static void GenerateSDKProjects()
+        {
+            // Create a copy of the packages as they might change after we create the MSBuild project
+            MakePackagesCopy();
+
+            Utilities.EnsureCleanDirectory(Application.dataPath.Replace("Assets", "MRTKBuild"));
+
+            string path = Application.dataPath.Replace("Assets", "MSBuild");
+            try
+            {
+                Utilities.EnsureCleanDirectory(path);
+            }
+            catch (IOException ex)
+            {
+                if (ex.Message.Contains(@".vs\MRTK\v15\Server\sqlite3\db.lock"))
```

Comment:
>.vs\MRTK\v15\Server\sqlite3\db.lock [](start = 42, length = 35)

Thoughts on just looking for db.lock specifically, - if only because seeing a version number makes me think this logic can get broken on future releases.

Reply:
Good idea.

---
In reply to: [301802469](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/5215#discussion_r301802469) [](ancestors = 301802469)